Terrain.java

```diff
@@ -1,24 +1,9 @@
-public abstract class Terrain {
-    private String _name;
-
-    // Default constructor
-    protected Terrain() {
-        this._name = "un-named";
-    }
-
-    // A Constructor that takes a name for the animal
-    protected Terrain(String name) {
-        this._name = name;
-    }
+public abstract class Terrain implements IDamageable {
```

Comment:
Your terains is "Damageable"?  ie. terrains have health?  If intended that would be interesting.
Also, you need the actually "implement" the interface by defing the required functions in this class.